compliance/virtualmachine/relay_test.go

```diff
@@ -0,0 +1,144 @@
+package virtualmachine
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	v1 "github.com/stackrox/rox/generated/internalapi/virtualmachine/v1"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestVMRelay(t *testing.T) {
+	suite.Run(t, new(relayTestSuite))
+}
+
+type relayTestSuite struct {
+	suite.Suite
+
+	ctx context.Context
+}
+
+func (s *relayTestSuite) SetupTest() {
+	s.ctx = context.Background()
+}
+
+func (s *relayTestSuite) TestHandleVsockConnection_InjectsVsockCID() {
+	conn := s.defaultVsockConn().withVsockCID(42)
+	client := newMockSensorClient()
+
+	err := handleVsockConnection(s.ctx, conn, client)
+	s.Require().NoError(err)
+
+	s.Equal("42", client.capturedRequests[0].IndexReport.VsockCid)
+
+}
+
+func (s *relayTestSuite) TestHandleVsockConnection_RejectsMalformedData() {
+	conn := s.defaultVsockConn().withData([]byte("malformed-data"))
+	client := newMockSensorClient()
+
+	err := handleVsockConnection(s.ctx, conn, client)
+	s.Error(err)
+}
+
+func (s *relayTestSuite) TestHandleVsockConnection_HandlesContextCancellation() {
+	conn := s.defaultVsockConn()
+	client := newMockSensorClient().withDelay(1 * time.Second)
+	ctx, cancel := context.WithTimeout(s.ctx, 100*time.Millisecond) // times out before sensor replies
+	defer cancel()
+
+	err := handleVsockConnection(ctx, conn, client)
+	s.Require().Error(err)
+	s.Contains(err.Error(), "context deadline exceeded")
+}
+
+func (s *relayTestSuite) TestReadFromConn_EnforcesSizeLimit() {
+	data := []byte("Hello, world!")
+
+	cases := map[string]struct {
+		sizeLimit   int
+		shouldError bool
+	}{
+		"data smaller than limit succeeds": {
+			sizeLimit:   2 * len(data),
+			shouldError: false,
+		},
+		"data of equal size as limit succeeds": {
+			sizeLimit:   len(data),
+			shouldError: false,
+		},
+		"data larger than limit fails": {
+			sizeLimit:   len(data) - 1,
+			shouldError: true,
+		},
+	}
+
+	conn := s.defaultVsockConn().withData(data)
+	connTimeout := 10 * time.Second // Not relevant in these tests
```

Comment:
Shouldn't this be inside the `s.Run`?

Reply:
Yeah, I would put everything that is needed for a test inside the `Run`. That would minimize the number of potential unforeseen interactions between the test-cases.